src/RA_Dlg_MemBookmark.cpp

```diff
@@ -20,437 +22,434 @@ namespace ra {
 
-                        if (nSelect == -1)
-                            break;
+                    if (nSelect == -1)
+                        break;
+                }
+                else if (pnmhdr->code == NM_DBLCLK)
+                {
+                    const NMITEMACTIVATE* pOnClick = reinterpret_cast<LPNMITEMACTIVATE>(pnmhdr);
+
+                    using namespace ra::rel_ops;
+                    if ((pOnClick->iItem != -1) && (pOnClick->iSubItem == SubItems::Desc))
+                    {
+                        nSelItemBM = pOnClick->iItem;
+                        nSelSubItemBM = pOnClick->iSubItem;
+
+                        EditLabel(pOnClick->iItem, pOnClick->iSubItem);
                     }
-                    else if (((LPNMHDR)lParam)->code == NM_DBLCLK)
+                    else if ((pOnClick->iItem != -1) && (pOnClick->iSubItem == SubItems::Address))
                     {
-                        hList = GetDlgItem(hDlg, IDC_RA_LBX_ADDRESSES);
-
-                        // const auto would give 'NMITEMACTIVATE* const', which isn't what we need
-                        const NMITEMACTIVATE* pOnClick = (LPNMITEMACTIVATE)lParam;
-
-                        using namespace ra::rel_ops;
-                        if ((pOnClick->iItem != -1) && (pOnClick->iSubItem == SubItems::Desc))
-                        {
-                            nSelItemBM = pOnClick->iItem;
-                            nSelSubItemBM = pOnClick->iSubItem;
-
-                            EditLabel(pOnClick->iItem, pOnClick->iSubItem);
-                        }
-                        else if ((pOnClick->iItem != -1) && (pOnClick->iSubItem == SubItems::Address))
-                        {
-                            g_MemoryDialog.SetWatchingAddress(m_vBookmarks[pOnClick->iItem]->Address());
-                            MemoryViewerControl::setAddress((m_vBookmarks[pOnClick->iItem]->Address() &
-                                ~(0xf)) - ((int)(MemoryViewerControl::m_nDisplayedLines / 2) << 4) + (0x50));
-                        }
+                        g_MemoryDialog.SetWatchingAddress(m_vBookmarks.at(pOnClick->iItem).Address());
+                        MemoryViewerControl::setAddress(
+                            (m_vBookmarks.at(pOnClick->iItem).Address() & ~(0xf)) -
+                            (ra::to_signed(MemoryViewerControl::m_nDisplayedLines / 2) << 4) + (0x50));
                     }
+                }
             }
-            return TRUE;
         }
-        case WM_COMMAND:
+        return LRESULT{TRUE};
+    };
+
+    // id->LOWORD(wParam), hwndCtl->lParam, codeNotify->HIWORD(wParam)
+    const auto OnCommand = [this](HWND hDlg, int id, [[maybe_unused]] HWND /*hwndCtl*/,
+                            [[maybe_unused]] UINT /*codeNotify*/) {
+        switch (id)
         {
-            switch (LOWORD(wParam))
+            case IDOK:
+                [[fallthrough]];
```

Comment:
explicit fallthrough comment is not necessary if no other code is associated with the case statement. it's clear that multiple cases will execute the same code.

Reply:
Sure it's fine, the compiler doesn't warn on fallthrough so I guess it doesn't really matter.